websites/K/Kahoot/presence.ts

```diff
@@ -38,153 +37,378 @@ async function getStrings() {
 			ndString: "kahoot.ndString",
 			rdString: "kahoot.rdString",
 			topX: "kahoot.topX",
+			onPodium: "kahoot.onPodium",
 			of: "kahoot.of",
-			questionNumber: "kahoot.questionNumber"
+			questionNumber: "kahoot.questionNumber",
+			feedback: "kahoot.feedback",
+			waitingAnswer: "kahoot.waitingAnswer",
+			drumRoll: "kahoot.drumRoll",
+			position: "kahoot.position",
+			teamTalk: "kahoot.teamTalk",
+			gameSummary: "kahoot.gameSummary",
+			login: "kahoot.login",
+			createHome: "kahoot.createHome",
+			discover: "kahoot.discover",
+			searchKahoots: "kahoot.searchKahoots",
+			kahootDetails: "kahoot.kahootDetails",
+			kahootProfile: "kahoot.kahootProfile",
+			myKahoots: "kahoot.myKahoots",
+			userReports: "kahoot.userReports",
+			myCourses: "kahoot.myCourses",
+			editingCourse: "kahoot.editingCourse",
+			viewingCourse: "kahoot.viewingCourse",
+			editingKahoot: "kahoot.editingKahoot",
+			previewingKahoot: "kahoot.previewingKahoot",
+			liveCourse: "kahoot.liveCourse",
+			liveCourseActivity: "kahoot.liveCourseActivity"
 		},
 		await presence.getSetting<string>("lang")
 	);
 }
 
-let currentQuestion: string,
-	score: string,
-	ranking: string,
-	top5: Promise<boolean>,
-	strings: { [key: string]: string },
+let strings: { [key: string]: string },
 	oldLang: string = null,
-	rankingSelector: HTMLElement;
+	browsingTimestamp = Math.floor(Date.now() / 1000),
+	// 0 - ready to be updated if needed
+	// 1 - updated, ready to be reset to 0
+	timestampUpdateState = 0;
 
 presence.on("UpdateData", async () => {
 	const presenceData: PresenceData = {
-			largeImageKey: "kahoot"
+			largeImageKey: "kahoot",
+			startTimestamp: browsingTimestamp
 		},
-		buttons = await presence.getSetting<boolean>("buttons"),
-		newLang = await presence.getSetting<string>("lang");
+		[buttons, newLang] = await Promise.all([
+			await presence.getSetting<boolean>("buttons"),
+			await presence.getSetting<string>("lang")
+		]);
 
 	oldLang ??= newLang;
 	strings ??= await getStrings();
 	if (oldLang !== newLang) oldLang = newLang;
```

Comment:
where do the strings get updated?

Reply:
This was also from the original codebase. I'll look into fixing this problem.